src/datetime_rotation.jl

```diff
@@ -2,9 +2,15 @@ using Dates
 import Base: isless
 
 raw"""
-    DatetimeRotatingFileLogger
+    DatetimeRotatingFileLogger(dir, file_pattern; always_flush=true)
 
-Constructs a FileLogger that rotates its file based on the current date.
+Construct a `DatetimeRotatingFileLogger` backed by a `SimpleLogger`.
+
+    DatetimeRotatingFileLogger(f::Function, dir, file_pattern; always_flush=true)
+
+Construct a `DatetimeRotatingFileLogger` backed by a `FormatLogger`.
```

Comment:
What if we left  whether it is backed by a `SimpleLogger` or a `FormatLogger`, or even wether it writes to disk itself, purely an implementation detail?
Then we can just document that it accepts a optional first argument which is a formatting function as per the FormatLogger.